### PR TITLE
fix: shoud not reject not better compact block

### PR DESCRIPTION
**Pull request by u2**
https://github.com/cryptape/ckb-internal/issues/1150

### CODE REVIEW

File: sync/src/relayer/tests/compact_block_process.rs [hunk 1]
```diff
@@ -144,50 +143,6 @@ fn test_unknow_parent() {
     );
 }
 
-#[test]
```
Comment (keroro520) [inline sync/src/relayer/tests/compact_block_process.rs]: I think this test case is still helpful.
So if we change to accept the not-better block, we should change the test case logic, but not remove it. Or add other tests instead.

**Comment by keroro520** [comment]
Relayer should only accept blocks with a height greater than `tip - N`. Otherwise, the attackers may launch a spam attack by sending many blocks with low difficulty.

**Comment by u2** [comment]
> Relayer should only accept blocks with a height greater than `tip - N`. Otherwise, the attackers may launch a spam attack by sending many blocks with low difficulty.

This another problem, create an issue?

File: sync/src/relayer/tests/compact_block_process.rs (hunk 1 above)
Comment (u2) [reply sync/src/relayer/tests/compact_block_process.rs]: updated.

File: sync/src/relayer/tests/compact_block_process.rs [hunk 2]
```diff
@@ -145,23 +144,23 @@ fn test_unknow_parent() {
...
+fn test_accept_not_a_better_block() {
     let (relayer, _) = build_chain(5);
-    let chain_state = relayer.shared.lock_chain_state();
-    let header = chain_state.tip_header();
+    let header = {
+        let chain_state = relayer.shared.lock_chain_state();
+        chain_state.tip_header().clone()
+    };
+
+    // The timestamp is random, so it may be not a better block.
```
Comment (keroro520) [inline sync/src/relayer/tests/compact_block_process.rs]: Can we give it less difficulty like [before](https://github.com/nervosnetwork/ckb/pull/1268/files#diff-0ae539c65d2a643736f1fdeb17c7cdb5L158) ?

Comment (u2) [reply sync/src/relayer/tests/compact_block_process.rs]: The less difficulty is an invalid header.

https://github.com/nervosnetwork/ckb/blob/develop/sync/src/relayer/compact_block_process.rs#L170

https://github.com/nervosnetwork/ckb/blob/develop/verification/src/header_verifier.rs#L149

Here it's a fork test(valid header but not a better block), not an invalid header test.

**Pull request merged**